src/entities/SubscriptionSet.ts

```diff
@@ -0,0 +1,99 @@
+import { Subscription } from './Subscription';
+import { SubscribeCapable, SubscriptionOptions } from './commonTypes';
+
+export class SubscriptionSet implements SubscribeCapable {
+  private channelNames: string[] = [];
+  private groupNames: string[] = [];
+  private options: SubscriptionOptions;
+  private pubnub: any;
+  private eventEmitter: any;
+  private subscriptionList: Subscription[];
+
+  constructor({
+    channels = [],
+    channelGroups = [],
+    subscriptionOptions,
+    eventEmitter,
+    pubnub,
+  }: {
+    channels: string[];
+    channelGroups: string[];
+    subscriptionOptions: SubscriptionOptions;
+    eventEmitter: any;
+    pubnub: any;
+  }) {
+    this.channelNames = [...this.channelNames, ...channels];
+    this.groupNames = [...this.groupNames, ...channelGroups];
```

Comment:
Wouldn't `this.channelNames` and `this.groupName` be empty during constructor call?

Reply:
updated!